include/swift/AST/Attr.def

```diff
@@ -51,6 +51,7 @@ TYPE_ATTR(autoclosure)
 TYPE_ATTR(convention)
 TYPE_ATTR(noescape)
 TYPE_ATTR(escaping)
+TYPE_ATTR(differentiable)
```

Comment:
Either this should be `_differentiable` or it should be behind the feature flag.

Reply:
Our initial thought was not to put this behind a build-script flag so that it can be tested in mainline builds.  Is it okay if we reject it programmatically in Sema based on the frontend feature flag?  We could reject it programmatically in the parser also.

Reply:
I added some code to reject it in Sema.